pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -93,6 +95,9 @@ type PipelineResult struct {
 // PipelineTask defines a task in a Pipeline, passing inputs from both
 // Params and from the output of previous tasks.
 type PipelineTask struct {
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
```

Comment:
If we had `Task *v1beta1.Task` instead of (or in addition to - for compatibility) the current `TaskSpec *TaskSpec`, then `metadata` would be included as optional - by design

...as how I understand from the `volumeClaimTemplate`.

Reply:
but this does not work for `TaskRef` if we want a "custom" metadata - but if we want to use the metdata as-is, I think it should work?

Reply:
thanks @jlpettersson, with `taskRef`, the referring task can have metadata which should be sufficient like we have today. My focus was to only add `metadata` with `taskSpec` as `pipeline` has no means of specifying it with `taskSpec`.